src/PrestaShopBundle/Routing/ControllerStack.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Routing;
+
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+
+/**
+ * Inspired by \Symfony\Component\HttpFoundation\RequestStack
+ */
+class ControllerStack
+{
+    /**
+     * @var Controller[]
+     */
+    private $controllers = [];
+
+    /**
+     * @param Controller $controller
+     */
+    public function push(Controller $controller)
+    {
+        $this->controllers[] = $controller;
+    }
+
+    /**
+     * @return Controller|null
+     */
+    public function pop()
+    {
+        if (!$this->controllers) {
```

Comment:
```suggestion
        if (empty($this->controllers)) {
```
or `!count()` since it's an array

Reply:
My code was inspired of Symfony 😄 
https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpFoundation/RequestStack.php

Reply:
:sob: Why they did this...

Reply:
I will fix this one and submit a PR on Symfony side :trollface: